src/lib/__tests__/routing-test.ts

```diff
@@ -0,0 +1,152 @@
+import { createSearchClient } from '../../../test/mock/createSearchClient';
+import { wait } from '../../../test/utils/wait';
+import historyRouter from '../routers/history';
+import instantsearch from '../..';
+import { connectSearchBox } from '../../connectors';
+
+describe('routing', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('writeOnDispose=true', () => {
+    test('cleans URL on dispose', async () => {
+      const pushState = jest.spyOn(window.history, 'pushState');
+      const writeDelay = 400;
```

Comment:
Do we need a `writeDelay` in every test? I understand that we want to test this behavior, but having it for every test (10x in this file) slows down the CI. Wondering if this is necessary for any other reason.